### PR TITLE
Added windows build support

### PR DESCRIPTION
**Pull request by brunnels**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -296,7 +299,7 @@ flash: all
 
 yui/$(YUI-COMPRESSOR):
 	$(Q) mkdir -p yui
-	cd yui; wget https://github.com/yui/yuicompressor/releases/download/v2.4.8/$(YUI-COMPRESSOR)
+	cd yui; wget --no-check-certificate https://github.com/yui/yuicompressor/releases/download/v2.4.8/$(YUI-COMPRESSOR) -O $(YUI-COMPRESSOR)
```
Comment (tve) [inline Makefile:295]: Why the insecure option? Does github serve up an invalid cert? And is the -O not default on windows?
Mhh, looks like no root certs come with the windows wget install, what a mess. I wouldn't want the insecure option on Linux, though...

**Comment by tve** [comment]
I removed the use of heatshrink, looks like your PR brings the entire submodule files in and also has a lot of changes related to heatshrink?

**Comment by tve** [comment]
What are the prereq's for building on windows?

**Comment by brunnels** [comment]
Most of the changes were so that it would build heatshrink.  If you're doing away with it then a lot of the changes are no longer required.  

mkespfsimage still needs mman-win32.  

I'll pull out all the code associated with heatshrink and add a win32 checking section to the Makefile so can keep the cert checking separate.

**Comment by brunnels** [comment]
I'll add win32 detection to the Makefile to handle the cert check concerns.

I'll remove everything concerning heatshrink and most of the changes will go away.

mman-win32 is still needed for mkespfsimage.

Other than that it's mainly visual studio support files.

**Comment by brunnels** [comment]
Okay, heatshrink is gone completely now and I modified the Makefile for mkespfsimage to support windows or linux

**Comment by brunnels** [comment]
Switched to master, will redo against master

**Pull request closed without merging**